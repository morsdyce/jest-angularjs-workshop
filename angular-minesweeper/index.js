import 'common/assets/stylesheets/minefield.css';

import App from 'angular-minesweeper/components/app/app.module';
import Common from 'angular-minesweeper/common/common.module';
import Board from 'angular-minesweeper/components/board/board.module';

export default angular.module('ngReact', [
  Common.name,
  Board.name,
  App.name
]);