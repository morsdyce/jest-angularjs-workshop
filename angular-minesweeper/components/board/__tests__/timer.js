import 'angular-minesweeper';
import { noop } from 'lodash';

describe('<timer>', () => {
  let createComponent, $scope, element, $interval;

  beforeEach(angular.mock.module('ngReact.board'));

  beforeEach(inject(($rootScope, $compile, _$interval_) => {
    $interval = _$interval_;

    createComponent = (html, scopeAttrs) => {
      $scope = angular.extend($rootScope.$new(), scopeAttrs);

      element = angular.element(html);
      element = $compile(element)($scope);
      $scope.$digest();
    };
  }));

  test('should render a timer', () => {
    const html = `<timer>`;
    createComponent(html, {});

    expect(element.html()).toMatchSnapshot();
  });

  test('should increase timer over time', () => {
    const html = `<timer>`;
    createComponent(html, {});

    expect(element.find('span').html()).toMatch("0");

    $interval.flush(1100);

    expect(element.find('span').html()).toMatch("1");
  });

  test('should reset timer on event reset-timer', () => {
    const html = `<timer>`;
    createComponent(html, {});

    expect(element.find('span').html()).toMatch("0");

    $interval.flush(1100);

    expect(element.find('span').html()).toMatch("1");

    $scope.$emit('reset-timer');
    $scope.$digest();

    expect(element.find('span').html()).toMatch("0");
  });

  test('should stop timer on event stop-game', () => {
    const html = `<timer>`;
    createComponent(html, {});

    expect(element.find('span').html()).toMatch("0");

    $interval.flush(1100);

    expect(element.find('span').html()).toMatch("1");

    $scope.$emit('game-over');
    $scope.$digest();

    expect(element.find('span').html()).toMatch("1");

    $interval.flush(1100);

    expect(element.find('span').html()).toMatch("1");

    $interval.flush(1100);

    expect(element.find('span').html()).toMatch("1");
  });

});

