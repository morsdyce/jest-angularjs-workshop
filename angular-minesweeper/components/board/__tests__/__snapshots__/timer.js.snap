// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<timer> should render a timer 1`] = `
"<div class=\\"timer number-window\\">
  <span class=\\"ng-binding\\">
    0
  </span>
</div>

"
`;
